lib/sequenceserver/blast/job.rb

```diff
@@ -20,23 +20,32 @@ def initialize(params)
         else
           validate params
           super do
-            @method    = params[:method]
+            @method = params[:method]
+            @query = params[:sequence]
             @qfile     = store('query.fa', params[:sequence])
             @databases = Database[params[:databases]]
             @advanced  = params[:advanced].to_s.strip
             @options   = @advanced + defaults
             # The following params are for analytics only
             @num_threads = config[:num_threads]
             @query_length = calculate_query_size
+            @number_of_sequences = calculate_number_of_sequences
```

Comment:
I think it would be more self explanatory if it was called number_of_query_sequences or sequences_in_query, what do you think?